audiomentations/augmentations/add_short_noises.py

```diff
@@ -305,30 +325,35 @@ def apply(self, samples, sample_rate):
 
             if noise_rms > 0:
                 if self.noise_rms in ["relative", "relative_to_whole_input"]:
-
                     desired_noise_rms = calculate_desired_noise_rms(
                         clean_rms, sound_params["snr_in_db"]
                     )
 
                     # Adjust the noise to match the desired noise RMS
                     noise_samples = noise_samples * (desired_noise_rms / noise_rms)
-
-                    noise_placeholder[
-                        start_sample_index:end_sample_index
-                    ] += noise_samples
-                if self.noise_rms == "absolute":
+                elif self.noise_rms == "absolute":
                     desired_noise_rms_db = sound_params["rms_in_db"]
                     desired_noise_rms_amp = convert_decibels_to_amplitude_ratio(
                         desired_noise_rms_db
                     )
                     gain = desired_noise_rms_amp / noise_rms
                     noise_samples = noise_samples * gain
 
-                    noise_placeholder[
-                        start_sample_index:end_sample_index
-                    ] += noise_samples
-        # Return a mix of the input sound and the added sounds
-        return samples + noise_placeholder
+                noise_placeholder[start_sample_index:end_sample_index] += noise_samples
+                if gain_signal:
+                    signal_mask[start_sample_index:end_sample_index] = np.maximum(
```

Comment:
noise_gain only takes positive values from my understanding. And signal_mask contains only 0.  The np.maximum is not so useful here, no ?

Reply:
Hmm, I think this is correct. It has to do with how we deal with the case when noises overlap. Let me try to illustrate

![bilde](https://user-images.githubusercontent.com/1470603/156788916-2ab1e971-0491-42fe-9a7d-2a2cca64fa61.png)

I'll do a few runs with plots and overlap to be sure

Reply:
Oh yes, I see, I haven't thought about that case !